supervisor/addons/addon.py

```diff
@@ -198,6 +198,11 @@ async def load(self) -> None:
         with suppress(DockerError):
             await self.instance.attach(version=self.version)
 
+        with suppress(DockerError):
+            image = self._image(self.data)
+            await self.instance.check_image(self.version, image, self.arch)
+            self.persist[ATTR_IMAGE] = image
```

Comment:
Doing this kind of action at the `load` is very early in the boot state. If there is no internet (which we not know as it's to early in the boot state) it can fail - today we still boot. We should make sure that we still boot without internet. Maybe it could make sense to to fix that after we loaded the NM and know if we can do this kind of operation or even as fixup after the system is booted and we have all system component loaded and working

Reply:
All the load methods are called during setup stage, not boot. Like that particular one is called as part of [this](https://github.com/home-assistant/supervisor/blob/a894c4589eed3c3ac03370d9d74284c1a6f4086b/supervisor/core.py#L149). I also checked before adding it and plugins, home assistant and addons are all loaded after dbus and host so that check should be functional by then.

Still that is why I added the suppress there just like we were doing before, to prevent failing startup over this.